pkg/cli/start/destroy.go

```diff
@@ -0,0 +1,248 @@
+package start
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/loft-sh/log"
+	"github.com/loft-sh/vcluster/pkg/platform/clihelper"
+	apiextensionsv1clientset "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/dynamic"
+)
+
+// define the order of resource deletion
+var resourceOrder = []string{
```

Comment:
Just stating the obvious, but this means we always have to keep the resources in sync here because we have to respect the ordering!?
Do you know what would happen if we would just delete all resources coming from `discoveryClient.ServerResourcesForGroupVersion("storage.loft.sh/v1")` with a Background deletion policy without a specific order?!

Reply:
Yeah, I think we need to do some e2e where we run vclusterctl commands against a platform instance.

I think the deletion would hang in most cases if we did an unordered delete because many things like accesskeys don't have finalizers